engine/kv_engine.cpp

```diff
@@ -1552,15 +1566,15 @@ Status KVEngine::StringSetImpl(const StringView& key, const StringView& value,
       pmem_allocator_->addr2offset(existing_record), key, value, expired_time);
 
   hash_table_->Insert(hint, ret.entry_ptr, StringDataRecord, new_record,
-                      PointerType::StringRecord);
+                      PointerType::StringRecord, entry_status);
   // Free existing record
   bool need_free =
       existing_record && ret.entry.GetRecordType() != StringDeleteRecord &&
       !ret.entry.IsExpiredStatus() /*Check if expired_key already handled by
                                        background cleaner*/
       ;
 
-  if (need_free) {
+  if (need_free && !ret.entry.IsExpiredStatus()) {
```

Comment:
!ret.entry.IsExpiredStatus() necessary?

Reply:
Yes, if background thread(CleanOutDated) pushed the expired key(OldDeletedRecord) into global cleaner, but haven't purge it. The foreground thread(Set) shouldn't push it again.